service/history/shard/context_impl.go

```diff
@@ -784,6 +784,10 @@ func (s *ContextImpl) GetCurrentExecution(
 	ctx context.Context,
 	request *persistence.GetCurrentExecutionRequest,
 ) (*persistence.GetCurrentExecutionResponse, error) {
+	if err := s.errorByState(); err != nil {
```

Comment:
any particular reason to check state for read-only methods now?

Reply:
There is no specific reason, simply trying to unify the behavior of IO to execution store.

Plus the impact is minor.